### PR TITLE
feat!: upgrade to synapse-sdk v1.0.1 and synapse-core v0.7.0

### LINKED ISSUE #402: Use Synapse's costs API
filecoin-pin's costs / runway calculations predate the new APIs in Synapse, but now they should be treated as canonical. There's a whole lot of code in src/core/payments/funding.ts that has our custom logic. Maybe we have to keep some or most of it? But if we can defer the tricky logic to synapse, even just synapse-core, then that would be preferred so we don't have to maintain our own and hit things like the new sybil fee for clients.

### LINKED ISSUE #559: migrate payments code for synapse-sdk GA costs reshape (synapse-sdk#823)
### Description

FilOzone/synapse-sdk#823 (~~open, unreleased~~) reshapes the Costs API for GA. Three filecoin-pin surfaces depend on what it removes or changes:

1. `getServicePrice()` is removed in favor of `getPriceList()`. Call sites: `src/payments/auto.ts:107`, `src/payments/status.ts:132`, `src/core/payments/index.ts:932`, `src/core/payments/funding.ts:437`. `src/core/payments/index.ts:50` also re-exports it, so the published core surface inherits the break (relevant to #279).
2. Operator-approval `maxLockupPeriod` moves from the hardcoded 30-day `LOCKUP_PERIOD` constant to a chain read (`getPriceList().lockups.defaultLockupPeriod`). `depositUSDFC` and `setServiceApprovals` pass an explicit 30-day value; that stays valid post-#823 but under-covers if a chain's default ever exceeds 30 days. `checkAllowances` mirrors the old `isFwssMaxApproved` comparison and will drift from the SDK's view; #823 adds a `requiredMaxLockupPeriod` option to `isFwssMaxApproved`, so the mirror can be deleted in favor of calling the SDK directly.
3. `calculateAdditionalLockupRequired` is reshaped; the funding paths that call it need adjusting.

### Impact

Nothing to do today: #823 is unmerged, blocked on contract redeploys, and the latest published SDK (0.41.0) matches our dependency. Once it ships, bumping `@filoz/synapse-sdk` / `@filoz/synapse-core` will not compile until these are migrated.

### Next steps

- Replace `getServicePrice()` call sites and the core re-export with `getPriceList()`.
- Drop the explicit `maxLockupPeriod` in `depositUSDFC` / `setServiceApprovals` and let the SDK resolve the chain default.
- Replace the comparison inside `checkAllowances` with the SDK's `isFwssMaxApproved`.

### PR DESCRIPTION
**Pull request by juliangruber**
## Summary

Upgrades to `@filoz/synapse-sdk` **v1.0.1** and `@filoz/synapse-core` **v0.7.0**, adopting the GA per-operation pricing model and the v1 API surface.

Part of #187 (filecoin-pin v1).

### Closes #559 — payments migration for the GA costs reshape (synapse-sdk#823)
- `getServicePrice()` → `getPriceList()` at every call site and the published `core/payments` re-export
- dropped the hardcoded `maxLockupPeriod` from `depositUSDFC` / `setServiceApprovals`, the SDK now resolves the chain default (`priceList.lockups.defaultLockupPeriod`)
- `checkAllowances` defers the "is it at max?" decision to the SDK's `isFwssMaxApproved`

### Closes #402 — use Synapse's costs API
- all pricing now flows from the on-chain price list (`getPriceList` / `storageInfo.pricing.priceList`) instead of the removed `USDFC_SYBIL_FEE` / `CDN_FIXED_LOCKUP` constants and the `minimumPricePerMonth` floor
- adopted the reshaped `calculateAdditionalLockupRequired` (takes the price list, the per-data-set floor is now the additive `datasetFeePerMonth`), and the funding planner now threads a `priceList` through instead of `pricePerTiBPerEpoch` + `minimumPricePerMonth`
- new-data-set funding now defers to synapse-core's `calculateUploadFees({ isNewDataSet: true })` and adds the `lifecycleReserveTarget` lockup, parity-tested against `calculateDepositNeeded`. This widens the planned top-up for new data sets to cover add-piece fees and the lifecycle reserve, not just the create-data-set fee and CDN lockup

### Other v1.0.1 breaking changes handled
- `getSizeFromPieceCID(cid)` → `from(cid).size` (PieceCID now carries `.size`)
- session-key `DeleteDataSetPermission` → `TerminateServicePermission`
- `storage.terminateDataSet()` → `storage.terminateService({ dataSetId, skipProvider: true })` — `skipProvider` preserves the existing direct-termination UX

## Follow-ups

These are intentionally out of scope.

- fuller deferral of the funding planner to synapse-core's `calculateDepositNeeded` / `getUploadCosts` (this PR only routes the new-data-set fee component through `calculateUploadFees`)
- adopting provider-relayed termination (the new v1 default) for the `data-set terminate` command

### CODE REVIEW

File: src/test/unit/payments-setup.test.ts [hunk 1]
```diff
@@ -45,6 +45,13 @@ vi.mock('@filoz/synapse-sdk', () => {
   }
 })
 
+// `checkAllowances` defers the "is this at max?" decision to the SDK helper.
+// Report "not max approved" so deposits take the approve-and-deposit path.
```
Comment (juliangruber) [inline src/test/unit/payments-setup.test.ts]:
```suggestion
// `checkAllowances` defers the maximum approved decision to the SDK helper.
// Report it's not, so deposits take the approve-and-deposit path.
```

File: src/payments/status.ts [hunk 2]
```diff
@@ -126,15 +125,14 @@ export async function showPaymentStatus(options: StatusOptions): Promise<void> {
...
+    const [accountSummary, storageInfo] = await Promise.all([
       synapse.payments.accountSummary({}),
       synapse.storage.getStorageInfo(),
-      getServicePrice(synapse.client),
     ])
     const runway = toStorageRunwaySummary(accountSummary)
 
     const pricePerTiBPerEpoch = storageInfo.pricing.noCDN.perTiBPerEpoch
-    const minimumPricePerMonth = servicePrice.minimumPricePerMonth
+    const minimumPricePerMonth = storageInfo.pricing.priceList.rates.datasetFeePerMonth
```
[the remark this answers is not in the document]
Comment (juliangruber) [reply src/payments/status.ts:136]: Done in 08101af — renamed the variable to `datasetFeePerMonth` to match the GA per-data-set fee model.

File: src/test/unit/payments-funding.test.ts [hunk 3]
```diff
@@ -231,11 +269,8 @@ describe('planFilecoinPayFunding', () => {
       synapse: makeSynapseStub() as any,
       targetRunwayDays: 10,
       pieceSizeBytes: 1024,
-      pricePerTiBPerEpoch: 1n,
-      minimumPricePerMonth: 60_000_000_000_000_000n,
     })
```
[the remark this answers is not in the document]
Comment (juliangruber) [reply src/test/unit/payments-funding.test.ts:296]: Done in 08101af — retitled to "adds create-data-set fees for new data sets in the shared funding plan".

File: src/test/unit/payments-funding.test.ts [hunk 4]
```diff
@@ -617,17 +650,17 @@ describe('computeAutoSetupTargetBalance', () => {
       filecoinPayBalance: 0n,
       availableFunds: 0n,
       copies,
-      minimumPricePerMonth,
+      priceList,
     })
     expect(result.requiredAvailableFunds).toBe(3_320_000_000_000_000_000n)
```
[the remark this answers is not in the document]
Comment (juliangruber) [reply src/test/unit/payments-funding.test.ts:713]: Done in 08101af — updated the comment to use "create-data-set fee" instead of "sybil fee".

File: upload-action/src/filecoin.js [hunk 5]
```diff
@@ -135,8 +133,7 @@ export async function handlePayments(synapse, options, logger) {
     allowWithdraw: false,
     targetRunwayDays: minStorageDays,
     pieceSizeBytes,
-    pricePerTiBPerEpoch: storageInfo.pricing.noCDN.perTiBPerEpoch,
-    minimumPricePerMonth: servicePrice.minimumPricePerMonth,
+    priceList: storageInfo.pricing.priceList,
     newDataSetCount,
     withCDN,
```
[the remark this answers is not in the document]
Comment (juliangruber) [reply upload-action/src/filecoin.js:139]: da862ab

File: src/data-set/run.ts [hunk 6]
```diff
@@ -308,7 +308,13 @@ export async function runTerminateDataSetCommand(dataSetId: number, options: Dat
...
     spinner.start('Submitting termination transaction...')
 
-    const txHash = await synapse.storage.terminateDataSet({ dataSetId: BigInt(dataSetId) })
+    // Direct (client-submitted) termination preserves the existing confirmation
+    // UX: `skipProvider` keeps the old `terminateDataSet` behaviour where the
+    // owner wallet submits the on-chain transaction and always gets a tx hash.
+    const { txHash } = await synapse.storage.terminateService({ dataSetId: BigInt(dataSetId), skipProvider: true })
+    if (txHash == null) {
+      throw new Error('Termination did not return a transaction hash')
+    }
```
Comment (SgtPooki) [inline src/data-set/run.ts:317]: i'm trying to remember what the tradeoff for users is with this.. if we skip provider, they can't release funds immediately and resolve things more quickly.. right?

do we want that to be the default in filecoin-pin?

File: upload-action/src/filecoin.js [hunk 7]
```diff
@@ -148,9 +145,10 @@ export async function handlePayments(synapse, options, logger) {
   }
 
   if (newDataSetCount > 0) {
+    const feeDescription = withCDN ? 'create-data-set fee and CDN lockup' : 'create-data-set fee'
```
Comment (SgtPooki) [inline upload-action/src/filecoin.js]: i think this wording is slightly out of date now.. the deposit would also cover lifecycle reserve and add-piece fees. not sure how we want to word this right now

**Review by SgtPooki — APPROVED** [review]
mostly lgtm. I pushed up a change to fix an edge-case on pricing with test: [`730fcda` (this PR)](https://github.com/filecoin-project/filecoin-pin/pull/574/commits/730fcda955dff5f40e655e77059b980427f1d83c)

File: src/data-set/run.ts (hunk 6 above)
Comment (juliangruber) [reply src/data-set/run.ts:317]: Intentional, to keep this PR behaviour-preserving. `skipProvider: true` reproduces the old `terminateDataSet` UX where the owner wallet submits the on-chain tx and we always get a tx hash back to confirm/wait on.

The v1 default (provider-relayed) is the better long-term path. I've left "adopt provider-relayed termination for `data-set terminate`" as a follow-up rather than flipping the default in this upgrade PR.

Or should we break now?

File: upload-action/src/filecoin.js (hunk 7 above)
Comment (juliangruber) [reply upload-action/src/filecoin.js]: a395e93

File: src/data-set/run.ts (hunk 6 above)
Comment (SgtPooki) [reply src/data-set/run.ts:317]: we are already breaking in this PR, right? if the question is "should we delay another breaking change outside of this breaking change" I would say no, if we're pretty confident we should do it in the future.

Comment (SgtPooki) [reply src/data-set/run.ts:317]: I think maybe we should add a tracking issue for adding an option to CLI & lib to do the old functionality (skip provider) and just do the "right" path now

**Review by SgtPooki — APPROVED** [review]

File: src/data-set/run.ts (hunk 6 above)
Comment (SgtPooki) [reply src/data-set/run.ts:317]: opened https://github.com/filecoin-project/filecoin-pin/issues/581 to keep track of this
